src/servers/grpc_server_v2.cc

```diff
@@ -1246,6 +1635,94 @@ InferResponseRelease(
   return nullptr;  // Success
 }
 
+template <typename TensorType>
+TRTSERVER_Error*
+ParseSharedMemoryParams(
+    const TensorType& tensor, bool* has_shared_memory, std::string* region_name,
+    int64_t* offset, size_t* byte_size)
+{
+  *has_shared_memory = false;
+  *offset = 0 /* default value */;
+  if (tensor.parameters().find("shared_memory_region") !=
```

Comment:
For perf reasons you should find each parameter only once and use the iterator thereafter (instead of using at() which causes another lookup in the map).